### PR TITLE
fix(firebase_auth): Fix `PlatformException` to `FirebaseAuthException` error message parsing

### PR DESCRIPTION
**Pull request by Isti01**
## Description

The message field of `FirebaseAuthException` contains broken JSON when receiving `BLOCKING_FUNCTION_ERROR_RESPONSE`
see (https://github.com/firebase/flutterfire/issues/11532)

## Related Issues

Fixes: firebase/flutterfire#11532

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel/utils/exception.dart [hunk 1]
```diff
@@ -58,9 +58,14 @@ FirebaseException platformExceptionToFirebaseAuthException(
       }
     }
 
+    var parsedMessage = platformException.message?.split(': ').last;
+    if (parsedMessage?.endsWith(' ]') ?? false) {
+      parsedMessage = parsedMessage!.substring(0, parsedMessage.length - 2);
+    }
```
Comment (russellwheatley) [inline packages/firebase_auth/firebase_auth_platform_interface/lib/src/method_channel/utils/exception.dart:64]:
```suggestion
    if (parsedMessage?.endsWith(' ]') ?? false) {
    	// Fixes JSON response from Auth blocking function: https://github.com/firebase/flutterfire/issues/11532
      parsedMessage = parsedMessage!.substring(0, parsedMessage.length - 2);
    }
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]
LGTM - just made a note suggestion

**Comment by russellwheatley** [comment]
FYI; I was able to reproduce this when I implemented a blocking function:
Screenshot 2023-09-01 at 10 49 49

**Comment by russellwheatley** [comment]
You need to run `melos run format` in the project to fix formatting CI. Install melos by running:
```
dart pub global activate melos
```

**Comment by Isti01** [comment]
Formatted the file, thanks!

**Comment by Isti01** [comment]
@russellwheatley Could you check again, please?

**Comment by russellwheatley** [comment]
Hey @Isti01 - sorry about the delay. do you mind resolving the conflicts and we can look to get this approved/merged? Thanks.

**Comment by Isti01** [comment]
Hey @russellwheatley, I resolved the conflicts

**Comment by Isti01** [comment]
Hi! @russellwheatley! I addressed the changes you requested but the merging is still blocked due to an unadressed change request, which is weird. Is that an issue?

**Review by russellwheatley — APPROVED** [review]

**Review by exaby73 — APPROVED** [review]

**Pull request merged**